### PR TITLE
Implemented categories for books and controller for that

### PR DESCRIPTION
**Pull request by Trayvor**

### CODE REVIEW

File: src/main/java/org/example/bookstore/dto/category/CategoryDto.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.example.bookstore.dto.category;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class CategoryDto {
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/dto/category/CategoryDto.java]: Is it response? If so, don't need to validate it

File: src/main/java/org/example/bookstore/model/Book.java [hunk 2]
```diff
@@ -5,17 +5,24 @@
...
 import lombok.Setter;
+import lombok.ToString;
 import org.hibernate.annotations.SQLDelete;
 import org.hibernate.annotations.SQLRestriction;
 
 @Getter
 @Setter
 @Entity
-@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Book.java]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
```

File: src/main/java/org/example/bookstore/model/Category.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    @Column
+    private String description;
+    @Column(nullable = false, name = "is_deleted", columnDefinition = "tinyint")
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Category.java]:
```suggestion
    @Column(nullable = false, columnDefinition = "tinyint")
```

File: src/main/java/org/example/bookstore/model/Category.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@SQLDelete(sql = "UPDATE categories SET is_deleted = true WHERE id=?")
+@SQLRestriction(value = "is_deleted=false")
+@Table(name = "categories")
+public class Category {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    @Column
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/model/Category.java]:
```suggestion
```

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -66,4 +66,11 @@ public List<BookDto> search(BookSearchParameters params) {
                 .map(bookMapper::toDto)
                 .toList();
     }
+
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java:69]: Should we set categories during saving and updating book?

File: src/main/java/org/example/bookstore/service/impl/CategoryServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    public CategoryDto save(CreateCategoryDto createCategoryDto) {
+        Category category = categoryMapper.toModel(createCategoryDto);
+        categoryRepository.save(category);
+        return categoryMapper.toDto(category);
+    }
+
+    @Override
+    public CategoryDto update(Long id, UpdateCategoryDto updateCategoryDto) {
+        Category category = categoryMapper.toModel(updateCategoryDto);
+        category.setId(id);
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/bookstore/service/impl/CategoryServiceImpl.java]:
```suggestion
        mapper.updateCategoryFromDto(categoryDto, category);
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java (hunk 5 above)
Comment (Trayvor) [reply src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java:69]: Why not?

Comment (Elena-Bruyako) [reply src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java:69]: Check please and text here, if during saving and updating book categories also set

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
just one comment

File: src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java (hunk 5 above)
Comment (Trayvor) [reply src/main/java/org/example/bookstore/service/impl/BookServiceImpl.java:69]: Sorry, missunderstanding

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**